api/hypershift/v1beta1/aws.go

```diff
@@ -109,8 +109,33 @@ type CapacityReservationOptions struct {
 	// +kubebuilder:validation:Enum:=OnDemand;CapacityBlocks
 	// +optional
 	MarketType MarketType `json:"marketType,omitempty"`
+
+	// preference specifies the preference for use of Capacity Reservations by the instance. Valid values include:
+	// "": No preference (platform default)
+	// "Open": The instance may make use of open Capacity Reservations that match its AZ and InstanceType
+	// "None": The instance may not make use of any Capacity Reservations. This is to conserve open reservations for desired workloads
+	// "CapacityReservationsOnly": The instance will only run if matched or targeted to a Capacity Reservation
+	//
+	// +kubebuilder:validation:Enum="";None;CapacityReservationsOnly;Open
+	// +optional
```

Comment:
do we need additional cel here? e.g:
can marketType=CapacityBlocks specify preferences?
can Tenancy=host/dedicated specify a marketType?
can Tenancy=host/dedicated specify preferences?

Reply:
added some validations